test/rules-tealium-fullstory.spec.ts

```diff
@@ -1,90 +1,79 @@
+      let testHarness: RulesetTestHarness;
+
+      beforeEach(async () => {
+        testHarness = await testEnv.createTestHarness(tealiumRules, {
+          utag: {
+            data: {
+              ...tealiumRetail,
+              tealium_event: '',
+            },
+          },
+        });
+      });
+
+      afterEach(async () => {
+        await testHarness.tearDown();
+      });
+
+      after(async () => {
+        await testEnv.tearDown();
+      });
+
+      it('monitors tealium_event', async () => {
+        await testHarness.execute(() => {
+          globalThis.utag.data.tealium_event = 'product_view';
+        });
+
+        const [id, payload] = await testHarness.popEvent();
+        expectEqual(id, 'product_view');
+        expectMatch(payload, tealiumRetail, 'product_id');
+        expectEqual(payload.order_total, '54.47');
+        expectEqual(payload.product_discount_amount, ['2.98']); // NOTE list reduced to single value
+        expectUndefined(payload, 'customer_first_name', 'customer_last_name', 'customer_email');
+      });
+
+      it('does not monitor properties not included in the source', async () => {
+        await testHarness.execute(() => {
+          utag.data.outsideScope = true;
+        });
+
+        await new Promise<void>((resolve, reject) => {
+          testHarness.popEvent(500)
+            .then(() => {
+              reject(new Error('Expected rejected promise due to no FS.event calls being present.'));
+            })
+            .catch(() => {
+              resolve();
+            });
+        });
+      });
     });
-
-    const [id, payload] = expectFS('event');
-    expectEqual(id, 'product_view');
-    expectMatch(payload, tealiumRetail, 'product_id');
-    expectEqual(payload.order_total, 54.47);
-    expectEqual(payload.product_discount_amount, 2.98); // NOTE list reduced to single value
-    expectUndefined(payload, 'customer_first_name', 'customer_last_name', 'customer_email');
-  });
-
-  it('should not monitor properties not included in the source', (done) => {
-    expectGlobal('utag').data.tealium_event = 'product_view';
-
-    ExpectObserver.getInstance().create({
-      // ensure this rule never reads on load
-      rules: [{ ...expectRule('fs-tealium-event'), readOnLoad: false }],
-    });
-
-    // NOTE this is a valid property to monitor
-    expectGlobal('utag').data.product_id = ['PROD789'];
-
-    // check the assignment
-    setTimeout(() => {
-      const [id, payload] = expectFS('event');
-      expectEqual(id, 'product_view');
-      expectEqual(payload.product_id, 'PROD789');
-    }, DataHandler.DefaultDebounceTime * 1.5);
-
-    // NOTE this is an invalid property to monitor because it is not picked
-    expectGlobal('utag').data.outsideScope = true;
-
-    // check the assignment
-    setTimeout(() => {
-      expectNoCalls(expectGlobal('FS'), 'event');
-      done();
-    }, DataHandler.DefaultDebounceTime * 1.5);
-  });
-
-  it('should identify user', () => {
```

Comment:
We don't make any `FS.identify` calls in the rules served with data layer capture integrations.